src/responder/nss/nss_protocol_pwent.c

```diff
@@ -293,13 +305,15 @@ nss_protocol_fill_pwent(struct nss_ctx *nss_ctx,
         SAFEALIGN_SET_STRING(&body[rp], gecos.str, gecos.len, &rp);
         SAFEALIGN_SET_STRING(&body[rp], homedir.str, homedir.len, &rp);
         SAFEALIGN_SET_STRING(&body[rp], shell.str, shell.len, &rp);
+        SAFEALIGN_SET_STRING(&body[rp], extra_data.data, extra_data.len, &rp);
```

Comment:
Why are you formatting this information as an additional structure, instead of simply adding these fileds one by one in the current structure ?